### PR TITLE
Add GH Actions.

### PR DESCRIPTION
**Pull request by billnapier**

### CODE REVIEW

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Python package
+
+on: [push]
```
Comment (Laptop765) [inline .github/workflows/python.yaml:3]: Optional but if you want the ability to click a button to run the workflow on demand (e.g. for testing) you can change this to:

```yaml
on:
  push:
  workflow_dispatch:
```

File: .github/workflows/python.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Lint with ruff
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          ruff --format=github --select=E9,F63,F7,F82 --target-version=py37 .
```
Comment (Laptop765) [inline .github/workflows/python.yaml:26]: Should `--target-version` match `python-version` from the `matrix` above?

**Review by Laptop765 — APPROVED** [review]

File: .github/workflows/python.yaml (hunk 1 above)
Comment (billnapier) [reply .github/workflows/python.yaml:3]: Cool!  I have so much to learn about GH Actions.

File: .github/workflows/python.yaml (hunk 2 above)
Comment (billnapier) [reply .github/workflows/python.yaml:26]: I think the idea is to match the minimum version from above, but I'm not 100% sure.  This is from their codelab, so for now I think I'll leave it until we have a compelling reason to change it.

**Pull request merged**